Containers/talk/start.sh

```diff
@@ -15,42 +15,47 @@ elif [ -z "$INTERNAL_SECRET" ]; then
     exit 1
 fi
 
+# TODO: check what happens if there is no IPv6
 set -x
-IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short)"
+if [ -n "$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_TALK="$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
+
+if [ -n "$(dig $NC_DOMAIN A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_NC="$(dig $NC_DOMAIN A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig $NC_DOMAIN AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_NC="$(dig $NC_DOMAIN AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
 set +x
 
 # Turn
-cat << TURN_CONF > "/etc/turnserver.conf"
-listening-port=$TALK_PORT
-fingerprint
-use-auth-secret
-static-auth-secret=$TURN_SECRET
-realm=$NC_DOMAIN
-total-quota=0
-bps-capacity=0
-stale-nonce
-no-multicast-peers
-simple-log
-pidfile=/var/tmp/turnserver.pid
-no-tls
-no-dtls
-userdb=/var/lib/turn/turndb
-# Based on https://nextcloud-talk.readthedocs.io/en/latest/TURN/#turn-server-and-internal-networks
-allowed-peer-ip=$IPv4_ADDRESS_TALK
-denied-peer-ip=0.0.0.0-0.255.255.255
-denied-peer-ip=10.0.0.0-10.255.255.255
-denied-peer-ip=100.64.0.0-100.127.255.255
-denied-peer-ip=127.0.0.0-127.255.255.255
-denied-peer-ip=169.254.0.0-169.254.255.255
-denied-peer-ip=172.16.0.0-172.31.255.255
-denied-peer-ip=192.0.0.0-192.0.0.255
-denied-peer-ip=192.0.2.0-192.0.2.255
-denied-peer-ip=192.88.99.0-192.88.99.255
-denied-peer-ip=192.168.0.0-192.168.255.255
-denied-peer-ip=198.18.0.0-198.19.255.255
-denied-peer-ip=198.51.100.0-198.51.100.255
-denied-peer-ip=203.0.113.0-203.0.113.255
-denied-peer-ip=240.0.0.0-255.255.255.255
+cat << TURN_CONF > "/etc/eturnal.yml"
+eturnal:
+  listen:
+    - ip: "::"
+      port: $TALK_PORT
+      transport: udp
+      enable_turn: true
+    - ip: "::"
+      port: $TALK_PORT
+      transport: tcp
+      enable_turn: true
+  log_dir: stdout
+  log_level: notice
+  realm: "$NC_DOMAIN"
+  secret: "$TURN_SECRET"
+  relay_ipv4_addr: "$IPv4_ADDRESS_NC"
+  relay_ipv6_addr: "$IPv6_ADDRESS_NC"
```

Comment:
I see you configure everything in the configuration file. Therefore, I would suggest the following changes in your Dockerfile:

Change this:
```
COPY --from=eturnal /usr/local/bin/eturnalctl /usr/local/bin/eturnalctl
COPY --from=eturnal /usr/local/bin/stun /usr/local/bin/stun
```
to
```
RUN ln -s $(find /opt/eturnal -name eturnalctl) /usr/local/bin/eturnalctl ; \
         ln -s $(find /opt/eturnal -name stun) /usr/local/bin/stun
```
(of course you can do it in the main RUN cmd)

Reply:
so opt also includes the same binary files?

Reply:
corret now?

Reply:
yes

Reply:
> so opt also includes the same binary files?

The one in our image is a bit extended for the STUN query, but you don't use it, so it is unneeded.